llvm/lib/ExecutionEngine/Interpreter/Execution.cpp

```diff
@@ -336,6 +336,16 @@ void Interpreter::visitICmpInst(ICmpInst &I) {
   Type *Ty    = I.getOperand(0)->getType();
   GenericValue Src1 = getOperandValue(I.getOperand(0), SF);
   GenericValue Src2 = getOperandValue(I.getOperand(1), SF);
+
+  if (Src1.IntVal.getBitWidth() < Src2.IntVal.getBitWidth())
```

Comment:
This cannot ever happen. The bitwidth of cmp operands are always identical.

Reply:
I don't understand. IntVal themselves store the bitwidth internally. How could they ever be mismatched? The IR itself would be broken.

Reply:
GenericValue here is calculated by the LLVM's Interpreter, it's possible that it doesn't conform completely with the IR (expecially regarding bit-witdh). I will reproduce and come back